### ISSUE TITLE
Makefiles not compatible with Cmder (a Windows console emulator) 

### ISSUE THREAD

**Issue by tdesveauxPKFX** [premake/premake-core#1102]
From #1091, Makefiles generated with Premake uses wrong mkdir command due to mis-identification of shell.

Here is a snippet that could replace the current identification that works with cmd.exe, Powershell, Cmder and Mingw.

```
SHELLTYPE := posix
ifeq (.exe,$(findstring .exe,$(ComSpec)))
  SHELLTYPE := msdos
endif
```

which would replace this:
https://github.com/premake/premake-core/blob/ff732af5f6cec5832d6e1e789ee8632ef2e32e7f/modules/gmake2/gmake2.lua#L275-L284

Just checking I'm not missing something before submitting a PR.

**Comment by samsinsane** [comment]
The [Cmder website](http://cmder.net/) says:
> you will have all Unix commands

The Unix `mkdir` supports `-p`, so this sounds like a bug in Cmder not Premake.

**Comment by tdesveauxPKFX** [comment]
Well, you have access to all Unix commands but as `mkdir` is present in both Unix and msdos, the behavior is to use msdos `mkdir`.
As an example, you can both use `del` and `rm` in Cmder.

**Comment by tdesveauxPKFX** [comment]
It's not that you don't have access to it but since there is a "conflict" due to the same name, the msdos command take priority.

It doesn't break Git Bash and shouldn't break Linux nor Unix in general.
I will submit a PR for this later.

**Comment by tdesveauxPKFX** [comment]
Fixed by #1103
